sdk/core/azure-core-amqp/inc/azure/core/amqp/network/amqp_header_detect_transport.hpp

```diff
@@ -8,22 +8,21 @@
 
 namespace Azure { namespace Core { namespace Amqp { namespace Network { namespace _internal {
 
-  /** A transport that detects the AMQP header on an incoming message. Used when listening for
-   * incoming AMQP connections and messages.
+  /** @brief Factory to create a transport which detects the AMQP header on an incoming message.
+   * Used when listening for incoming AMQP connections and messages.
    */
-  class AmqpHeaderDetectTransport final : public Transport {
+  class AmqpHeaderDetectTransportFactory final {
 
   public:
-    /** Construct a new instance of the Amqp Header Detect Transport.
-     * 	 *
-     * 	 * @param parentTransport The parent transport to read from.
-     * 	 * @param eventHandler The event handler to notify when the AMQP header is detected.
-     * 	 *
+    /** Construct a new instance of an Amqp Header Detect Transport.
+     *
+     * @param parentTransport The parent transport to read from.
+     * @param eventHandler The event handler to notify when the AMQP header is detected.
+     *
      */
-    AmqpHeaderDetectTransport(
+    static Transport Create(
```

Comment:
I see the `Transport` is now final, if it is ever possible for it to become non-`final` again, change the return type into `unique_ptr<Transport>`. Otherwise, is it really necessary for there to be a factory - maybe just having a public constructor in `Transport` class is enough?

Reply:
If you want to expose azure-c-shared-utility and uAMQP constructs outside the implementation, we can remove the factory.

But I'm trying very hard to avoid exposing uAMQP functionality outside this codebase. Making the transport type functions be factories for transport objects allows us to hide the actual uAMQP underpinnings (the uAMQP goo gets hidden in the factory).